core/src/main/java/hudson/TcpSlaveAgentListener.java

```diff
@@ -301,6 +325,27 @@ private void error(PrintWriter out, String msg) throws IOException {
         }
     }
 
+    // This is essentially just to be able to pass the parent thread into the callback, as it can't access it otherwise
+    private abstract class ConnectionHandlerFailureCallback {
+        private Thread parentThread;
+        private int port;
+
+        public ConnectionHandlerFailureCallback(Thread parentThread, int port) {
+            this.parentThread = parentThread;
+            this.port = port;
+        }
+
+        public int getPort() {
```

Comment:
I see no usages of this `port` entry. Maybe makes sense to just remove that

Reply:
Ah yeah, remnant from before.  Fixed.